numpy/core/fromnumeric.py

```diff
@@ -2120,6 +2122,19 @@ def sum(a, axis=None, dtype=None, out=None, keepdims=np._NoValue,
     >>> np.sum([])
     0.0
 
+    For floating point numbers the numerical precision of sum (and
+    ``np.add.reduce``) is in general limited by directly adding each number
+    individually to the result causing rounding errors in every step.
+    However, often numpy will use a  numerically better approach
+    (pairwise summation) leading to improved precision in many use cases.
+    This improved precision is always provided when no ``axis`` is given.
+    When ``axis`` is given, it will depend on which axis is summed.
+    Technically, to provide the best speed possible, the improved precision
+    is only used when the summation is along the fast axis in memory.
+    Note that the exact precision may vary depending on other parameters.
+    In contrast to NumPy, Python's ``math.fsum`` function uses a slower but
+    more precise approach to summation.
```

Comment:
Is it worth adding that for lower-precision floats such as `f4`, one can pass in `dtype='f8'` to increase precision?

Reply:
I guess so, although it is growing so long... Maybe the real solution would be to have a longer paragraph about floating point rounding in the user guide and link that...